### DISCUSSION TITLE
Handling an S3Event as a Lambda destination function

### DISCUSSION THREAD

**Discussion by simonmurdock** [aws/aws-lambda-dotnet#1294] [Q&A] [answered]
I have a primary Lambda function which is able to parse an `S3Event` fine, however when it completes and passes to the next lambda function, that one receives an invocation event of some type which looks like this:

```json
{
    "version": "1.0",
    "timestamp": "2019-11-14T18:16:05.568Z",
    "requestContext": {
        "requestId": "e4b46cbf-b738-xmpl-8880-a18cdf61200e",
        "functionArn": "arn:aws:lambda:us-east-2:123456789012:function:my-function:$LATEST",
        "condition": "RetriesExhausted",
        "approximateInvokeCount": 3
    },
    "requestPayload": {
        // normal S3Event here
    },
    "responseContext": {
        "statusCode": 200,
        "executedVersion": "$LATEST",
        "functionError": "Unhandled"
    },
    "responsePayload": {
        "errorMessage": "RequestId: e4b46cbf-b738-xmpl-8880-a18cdf61200e Process exited before completing request"
    }
}
```

Where is this type stored in the Amazon.Lambda.RuntimeSupport libraries to let me cast it directly in my function handler? at the moment I have to cast it to a custom type with RequestPayload being an `S3Event`.

**Comment by ashishdhingra** [comment, marked as answer]
@simonmurdock Looks like you are getting response JSON for function invocation errors. Kindly refer example JSON at [Asynchronous invocation](https://docs.aws.amazon.com/lambda/latest/dg/invocation-async.html). There is no type in the Lambda libraries which map to these invocation errors, you could create your custom implementation to model this payload.

Thanks,
Ashish

**Reply by simonmurdock** [reply]
If there is no type then that is fine, It would be a neat addition, so we can simply cast the response.
